### PR TITLE
feat(tile-converter): conversion progress

### PR DESCRIPTION
**Pull request by mspivak-actionengine**

### CODE REVIEW

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 1]
```diff
@@ -139,6 +139,9 @@ export default class I3SConverter {
     meshTopologyTypes: new Set(),
     metadataClasses: new Set()
   };
+  tileCountTotal: number = 0;
```
Comment (ibgreen) [inline modules/tile-converter/src/i3s-converter/i3s-converter.ts:142]: Nit: Add a TSDoc in front of each field. Then when hovering over these in vscode the documentation is shown.
```suggestion
  /** Total count of tiles in tileset */
  tileCountTotal: number = 0;
```

**Review by ibgreen — APPROVED** [review]

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 2]
```diff
@@ -596,7 +603,14 @@ export default class I3SConverter {
...
     if (sourceTile.id) {
-      console.log(`[convert]: ${sourceTile.id}`); // eslint-disable-line
+      // Print the tile number that is currently being converted.
+      this.tileCountCurrentlyConverting++;
+      let percentString = '';
+      if (this.tileCountTotal) {
+        const percent = (this.tileCountCurrentlyConverting / this.tileCountTotal) * 100;
+        percentString = percent.toFixed(this.numberOfDigitsInPercentage);
+      }
+      console.log(`[convert]:[${percentString}%]: ${sourceTile.id}`); // eslint-disable-line
```
Comment (belom88) [inline modules/tile-converter/src/i3s-converter/i3s-converter.ts:613]: I would make more compact style:
```suggestion
      console.log(`[convert ${percentString}%]: ${sourceTile.id}`); // eslint-disable-line
```

**Review by belom88 — CHANGES_REQUESTED** [review]

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts (hunk 1 above)
Comment (mspivak-actionengine) [reply modules/tile-converter/src/i3s-converter/i3s-converter.ts:143]: Done

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 3]
```diff
@@ -344,7 +350,8 @@ export default class I3SConverter {
       return null;
     }
     if (sourceTile.id) {
-      console.log(`[analyze]: ${sourceTile.id}`); // eslint-disable-line
+      this.tileCountTotal++;
+      console.log(`[analyze]:[${this.tileCountTotal}]: ${sourceTile.id}`); // eslint-disable-line
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply modules/tile-converter/src/i3s-converter/i3s-converter.ts]: Done

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts (hunk 2 above)
Comment (mspivak-actionengine) [reply modules/tile-converter/src/i3s-converter/i3s-converter.ts]: Done

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 4]
```diff
@@ -596,7 +610,14 @@ export default class I3SConverter {
       return traversalProps;
     }
     if (sourceTile.id) {
-      console.log(`[convert]: ${sourceTile.id}`); // eslint-disable-line
+      // Print the tile number that is currently being converted.
+      this.tileCountCurrentlyConverting++;
+      let percentString = '';
+      if (this.tileCountTotal) {
+        const percent = (this.tileCountCurrentlyConverting / this.tileCountTotal) * 100;
+        percentString = ' ' + percent.toFixed(this.numberOfDigitsInPercentage);
```
Comment (belom88) [inline modules/tile-converter/src/i3s-converter/i3s-converter.ts:618]:
```suggestion
        percentString = ' ' + percent.toFixed(this.numberOfDigitsInPercentage);
```
```suggestion
        percentString = ` ${percent.toFixed(this.numberOfDigitsInPercentage)}`;
```

**Review by belom88 — APPROVED** [review]

**Pull request merged**